wandb/sdk/artifacts/artifact.py

```diff
@@ -1145,7 +1145,7 @@ def __setitem__(self, name: str, item: WBValue) -> ArtifactManifestEntry:
     @contextlib.contextmanager
     @ensure_not_finalized
     def new_file(
-        self, name: str, mode: str = "w", encoding: str | None = None
+        self, name: str, mode: str = "x", encoding: str | None = None
     ) -> Iterator[IO]:
```

Comment:
note: Changing default `mode` from `"w"` to `"x"` is technically a **breaking change**, so open to alternatives or reconsidering.

At the same time, it better describes the current default behavior of this method, since:
- filemode `"w"` (current default, and probably more common in most Python code) communicates that existing files (if any) will be overwritten
  - ⬆️ This is **not** the current behavior 
- filemode `"x"` (new default) communicates "exclusive" write mode, ie the user should expect an error if the file already exists
  - ⬆️ This is the **actual** current behavior

Reply:
If I understand correctly, the only change in behavior is that we now properly support the "w" mode, right? So users who didn't specify the mode or who specified "x" will not observe any difference?

Might be worth mentioning in the changelog in the Fixed section!